rfq/order.go

```diff
@@ -167,6 +184,29 @@ func (c *AssetSalePolicy) CheckHtlcCompliance(
 	return nil
 }
 
+// TrackAcceptedHtlc accounts for the newly accepted htlc. This may affect the
+// acceptance of future htlcs.
+func (c *AssetSalePolicy) TrackAcceptedHtlc(htlc lndclient.InterceptedHtlc) {
+	c.CurrentAmountMsat += htlc.AmountOutMsat
+
+	htlcIDStr := htlcIdentifierStr(
+		htlc.IncomingCircuitKey.ChanID.ToUint64(),
+		htlc.IncomingCircuitKey.HtlcID,
+	)
+
+	c.htlcToAmt.Store(htlcIDStr, htlc.AmountOutMsat)
+}
+
+// UntrackHtlc stops tracking the uniquely identified htlc.
+func (c *AssetSalePolicy) UntrackHtlc(htlcIDStr string) {
+	amt, found := c.htlcToAmt.LoadAndDelete(htlcIDStr)
+	if !found {
+		return
+	}
+
+	c.CurrentAmountMsat -= amt
```

Comment:
Yeah, we definitely need a mutex here. Otherwise another goroutine might read this value while we're waiting on the `LoadAndDelete` above to complete.